lib/request.js

```diff
@@ -1,19 +1,51 @@
 'use strict'
 
 const { Readable } = require('readable-stream')
-const Url = require('url')
+const qs = require('querystring')
 const util = require('util')
+const url = require('url')
+
+function parseURL (options) {
+  const parsedURL = url.parse(
```

Comment:
As far as I know, `url.parse` is deprecated, you should use WHATWG URL API instead :)

Reply:
You should use `const { URL} = require('url')` and then `new URL()`.

Reply:
The new WHATWG `URL` changes a lot. 

For example, `'/example?q=1'` is an invalid URL now:

```
new URL('/example?q=1')
{ TypeError [ERR_INVALID_URL]: Invalid URL: /example?q=1
    at onParseError (internal/url.js:240:17)
    at parse (internal/url.js:249:3)
    at new URL (internal/url.js:324:5) input: '/example?q=1' }
```

As a result, a `base` option may be required.

So, I thinks it's another task to mitigate to the new WHATWG `URL`.

Reply:
I agree